src/helpViewer/helpProvider.ts

```diff
@@ -270,33 +268,54 @@ export class AliasProvider {
     }
 
     // call R script `getAliases.R` and parse the output
-    private getAliasesFromR(): undefined | AllPackageAliases {
-
-        // get from R
-		const lim = '---vsc---'; // must match the lim used in R!
-		const re = new RegExp(`^.*?${lim}(.*)${lim}.*$`, 'ms');
-        const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'vscode-R-aliases'));
-        const tempFile = path.join(tempDir, 'aliases.json');
-        const cmd = `"${this.rPath}" --silent --no-save --no-restore --slave -f "${this.rScriptFile}" > "${tempFile}"`;
-
-        let allPackageAliases: undefined | AllPackageAliases = undefined;
-        try{
-            // execute R script 'getAliases.R'
-            // aliases will be written to tempFile
-            cp.execSync(cmd, { cwd: this.cwd });
-
-            // read and parse aliases
-            const txt = fs.readFileSync(tempFile, 'utf-8');
-            const json = txt.replace(re, '$1');
-            if(json){
-                allPackageAliases = <{[key: string]: PackageAliases}> JSON.parse(json) || {};
+    private async getAliasesFromR(): Promise<undefined | AllPackageAliases> {
+        const lim = '---vsc---';
+        const options: cp.ExecSyncOptionsWithStringEncoding = {
+            cwd: this.cwd,
+            encoding: 'utf-8',
```

Comment:
I think this encoding is ignored and the type could be changed to `cp.CommonOptions`

Reply:
Done thanks!